### PR TITLE
[fix] exists_movie support movie same name different year

### PR DESCRIPTION
**Pull request by vxcamiloxv**
### Motivation for changes:
Currently `exists_movie` has a bug on movies with the same name but different year, there cases like `The Lion King (1994)` and  `The Lion King (2019)` and currently `exists_movie` reject it due has the same name, this PR fix that bug

### Detailed changes:
- add the year of the movie to the key identifier to allow movie with same neam and different year

### Addressed issues:
- No open issue yet

### Log and/or tests output (preferably both):
```
adding: The Lion King 1994
adding: Pokémon Detective Pikachu 2019
adding: Ant-Man and the Wasp 2018
Teen Titans Go! To the Movies 2018
2019-10-14 12:34 VERBOSE  task          copy-movies     REJECTED: `Teen Titans Go! To the Movies (2018)[1080p]` by exists_movie plugin because movie exists
Steven Universe: The Movie 2019
2019-10-14 12:34 VERBOSE  task          copy-movies     REJECTED: `steven.universe.the.movie.2019.hdrip.720p.latino` by exists_movie plugin because movie exists
The Lion King 2019
El Camino: A Breaking Bad Movie 2019
2019-10-14 12:34 VERBOSE  task          copy-movies     REJECTED: `El.camino.a.breaking.bad.movie.2019.1080p-dual-lat-cinecalidad.is` by exists_movie plugin because movie exists
2019-10-14 12:34 VERBOSE  details       copy-movies     Summary - Accepted: 1 (Rejected: 19 Undecided: 0 Failed: 0)
2019-10-14 12:34 INFO     copy          copy-movies     Creating destination directory `/movies/The Lion King (2019)`
```

### CODE REVIEW

File: flexget/plugins/filter/exists_movie.py [hunk 1]
```diff
@@ -162,28 +165,36 @@ def on_task_filter(self, task, config):
             count_entries += 1
             log.debug('trying to parse entry %s' % entry['title'])
             if config.get('lookup') == 'imdb':
-                key = 'imdb_id'
-                if not entry.get('imdb_id', eval_lazy=False):
+                keyImdb = 'imdb_id'
```
Comment (liiight) [inline flexget/plugins/filter/exists_movie.py]: please stick to snake_case. Also, if this isn't used anywhere else, I really don't see the point of assigning into variable.

File: flexget/plugins/filter/exists_movie.py [hunk 2]
```diff
@@ -162,28 +165,36 @@ def on_task_filter(self, task, config):
...
                         imdb_lookup.lookup(entry)
                     except plugin.PluginError as e:
                         log.trace('entry %s imdb failed (%s)' % (entry['title'], e.value))
                         incompatible_entries += 1
                         continue
+                key = entry[keyImdb]
             else:
-                key = 'movie_name'
-                if not entry.get('movie_name', eval_lazy=False):
+                keyName = 'movie_name'
```
Comment (liiight) [inline flexget/plugins/filter/exists_movie.py]: please use snake case

File: flexget/tests/test_exists_movie.py [hunk 3]
```diff
@@ -120,6 +130,19 @@ def test_existing_files(self, execute_task):
             'accepted', title='Gone.Missing.2013'
         ), 'Gone.Missing.2013 should have been accepted'
 
+    def test_same_name_diff_year(self, execute_task):
```
Comment (liiight) [inline flexget/tests/test_exists_movie.py]: nice test. Maybe add another one as a negative test, where exist movie rejects a movie with the same year?

File: flexget/plugins/filter/exists_movie.py (hunk 2 above)
Comment (vxcamiloxv) [reply flexget/plugins/filter/exists_movie.py]: done https://github.com/Flexget/Flexget/pull/2455/commits/74f67b9bb953d55406ffa9647fcf20d4f26a2196

File: flexget/tests/test_exists_movie.py (hunk 3 above)
Comment (vxcamiloxv) [reply flexget/tests/test_exists_movie.py]: I already added that expect https://github.com/Flexget/Flexget/pull/2455/files#diff-e1ead679e7478a7d54927d28c3a18961R136

**Comment by liiight** [comment]
Thanks for this!

**Pull request merged**